### ISSUE TITLE
Fix S1854 FP: Unary increment prefix operator is not a useless assignment.

### ISSUE THREAD

**Issue by nrankin18** [SonarSource/sonar-dotnet#8501]
### Description

S1854 throws a false positive when using a unary prefix operator on a variable that is passed into a function.

### Repro steps

```cs
int foo = 1;
bar(--foo); // S1854 is thrown
```

### Expected behavior

S1854 should not be thrown as the prefix operator alters the value passed into a function.

### Actual behavior

S1854 is thrown.

### Known workarounds

None.

### Related information

* C# version: 10.0
* Visual Studio version: 17.8.2
* dotnet version : net6.0
* Operating System: Windows 11 Home 10.0.22000 Build 22000

**Comment by pavel-mikula-sonarsource** [comment]
Hi @nrankin18,

this is expected unless you use the decremented value of `foo` later in your method.

We have a related ticket here #8330 to provide better explanation in this scenario.
